aea/protocols/base.py

```diff
@@ -292,3 +299,23 @@ def from_dir(cls, directory: str) -> "Protocol":
         )
         protocol = Protocol(protocol_id, serializer_class(), protocol_config)
         return protocol
+
+    def setup(self, *args, **kwargs):
```

Comment:
maybe, load would be a better name? Also avoids confusion with the setup/teardown methods in the skill modules

Reply:
yes